docs/rules/no-domains.md

```diff
@@ -0,0 +1,37 @@
+# No Domains
+
+Rule `no-domains` will enforce that domains are not used within urls.
```

Comment:
Only a slight thought that maybe this should be `no-url-domains` to keep it in sync with the rule it shares functionality with. I'm completely open if you prefer this though, just a thought.